lib/guest_installation_and_configuration_base.pm

```diff
@@ -1730,15 +1735,18 @@ sub start_guest_installation {
     my $_guest_installation_dryrun_log = "$common_log_folder/$self->{guest_name}/$self->{guest_name}" . "_installation_dryrun_log_" . $_start_installation_timestamp;
     my $_guest_installation_log = "$common_log_folder/$self->{guest_name}/$self->{guest_name}" . "_installation_log_" . $_start_installation_timestamp;
     assert_script_run("touch $_guest_installation_log && chmod 777 $_guest_installation_log");
-    if (script_run("set -o pipefail;$self->{virt_install_command_line_dryrun} 2>&1 | tee -a $_guest_installation_dryrun_log", timeout => 600 / get_var('TIMEOUT_SCALE', 1)) ne 0) {
-        record_info("Guest $self->{guest_name} installation dry run failed", "The virt-install command used is $self->{virt_install_command_line_dryrun}");
-        $self->record_guest_installation_result('FAILED');
+    # Dry run always timeout when downloading initrd from download.opensuse.org in O3
+    my $ret = script_run("set -o pipefail; $self->{virt_install_command_line_dryrun} 2>&1 | tee -a $_guest_installation_dryrun_log", timeout => 600 / get_var('TIMEOUT_SCALE', 1), die_on_timeout => 0);
+    save_screenshot;
+    unless (defined(script_run('set +o pipefail', die_on_timeout => 0))) {
+        reconnect_when_ssh_console_broken;
```

Comment:
Why add reconnect_when_ssh_console_broken here ?

Reply:
because ssh console is broken when virt-install timeouts.  fg. https://openqa.opensuse.org/tests/3106501#step/unified_guest_installation/304. we have to reconnect ssh console to continue tests.

Reply:
I do not think this is quite necessary, because there is "check_root_ssh_console" in post_fail_hook in concurrent_guest_installation. If set +o pipefail fails, just let it go. So remove this reconnect_when_ssh_console_broken ?

Reply:
No, reconnect is not just for post_fail_hook. Test should be continued when ssh console is broken, because other guests are expected to be installed as well.